### ISSUE TITLE
Wrap the new Filter class to support the new where methods

### ISSUE THREAD

**Issue by jmao-denver** [deephaven/deephaven-core#1613]
The db-module-refactor project has improved the table filter ops. This potentially requires that the new hand-crafted Python package to wrap the Filter interface.

**Comment by jmao-denver** [comment]
Ryan:
```
We want to dump where_one_of
and tell people to use structured filter expressions if they have complex things to express

FilterAnd.of(FilterOr.of(Filter.of("true"), Filter.of("false")), Filter.of("column"))
```
Jianfeng:
```
so the wrapping should solve the ugliness of the above expression?
```
Ryan:
```
Or just make it do-able from Python.
```

**Comment by jmao-denver** [comment]
closed by #2336
